### PR TITLE
Feature: Cached family and  funds errors (kids-1274)

### PR DESCRIPTION
**Pull request by Daniela510**

### CODE REVIEW

File: lib/features/children/add_member/pages/failed_allowances_bottomsheet.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      icon: Stack(
+        alignment: Alignment.center,
+        children: [
+          const Padding(
+            padding: EdgeInsets.all(8),
+            child: CircleAvatar(
+              radius: 60,
+              backgroundColor: FamilyAppTheme.primary95,
+            ),
+          ),
```
Comment (Daniela510) [inline lib/features/children/add_member/pages/failed_allowances_bottomsheet.dart:36]: This one is unfinished, no design yet.

File: lib/features/children/add_member/pages/failed_allowances_bottomsheet.dart [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+class InsufficientFundsBottomsheet extends StatelessWidget {
+  InsufficientFundsBottomsheet({required this.amount, super.key});
+  final int amount;
+  final stripeCubit = getIt<StripeCubit>();
+
+  @override
+  Widget build(BuildContext context) {
+    return GivtBottomSheet(
+      title: "You had inssufficient funds",
```
Comment (TammiLion) [inline lib/features/children/add_member/pages/failed_allowances_bottomsheet.dart:26]: inssufficient => insufficient

Comment (TammiLion) [reply lib/features/children/add_member/pages/failed_allowances_bottomsheet.dart:26]: Ah i see now design yet, you can ignore

File: lib/features/children/add_member/pages/failed_vpc_bottomsheet.dart [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                              backgroundColor: FamilyAppTheme.primary95,
+                            ),
+                          ),
+                          walletEmptyIcon(width: 140, height: 140),
+                        ],
+                      ),
+                content: (state.status == CachedMembersStateStatus.loading)
+                    ? const SizedBox.shrink()
+                    : BodyMediumText(
+                        'We couldn’t take the \$0.50 for verification and the \$${walletAmount.toStringAsFixed(0)} for your child’s wallet.\n\nCheck your payment details and try again or choose another one.',
```
Comment (TammiLion) [inline lib/features/children/add_member/pages/failed_vpc_bottomsheet.dart:73]: Do we perhaps need `child's wallet` versus `childrens wallets` here?

File: lib/features/children/add_member/pages/failed_vpc_bottomsheet.dart [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+              );
+            },
+          );
+        },
+      ),
+    );
+  }
+
+  static void show(
+      BuildContext context, List<Member> cachedMembers, VoidCallback onThen) {
```
Comment (TammiLion) [inline lib/features/children/add_member/pages/failed_vpc_bottomsheet.dart:133]: can we have a more describing name than "onThen" pwetty pwease 🥹 something like onBottomSheet... (close, open etc)

File: lib/features/family/features/profiles/cubit/profiles_cubit.dart [hunk 5]
```diff
@@ -41,6 +46,18 @@ class ProfilesCubit extends Cubit<ProfilesState> {
         fetchActiveProfile();
       },
     );
+    _cachedMembersSubscription =
+        _cachedMembersRepository.onCachedMembersChanged().listen(
+      (members) {
+        emit(
```
Comment (TammiLion) [inline lib/features/family/features/profiles/cubit/profiles_cubit.dart:52]: I would create a method for this. You can reuse that down below as well.

File: lib/features/family/features/profiles/cubit/profiles_state.dart [hunk 6]
```diff
@@ -4,7 +4,9 @@ abstract class ProfilesState extends Equatable {
   const ProfilesState({
     required this.profiles,
     required this.activeProfileIndex,
+    this.cachedMembers = const [],
```
Comment (TammiLion) [inline lib/features/family/features/profiles/cubit/profiles_state.dart:7]: I think in the PR you are only putting this in when calling the ProfilesUpdatedState. So I would not make it a member of ProfileState. Just of the ProfilesUpdatedState.

File: lib/features/family/features/profiles/screens/profile_selection_screen.dart [hunk 7]
```diff
@@ -86,8 +94,8 @@ class _ProfileSelectionScreenState extends State<ProfileSelectionScreen> {
             isError: true,
           );
         } else if (state is ProfilesNotSetupState) {
-          if (CachedFamilyUtils.isFamilyCacheExist()) {
-            await context.pushNamed(FamilyPages.cachedChildrenOverview.name);
+          if (state.cachedMembers.isNotEmpty && !isCachedMembersBottomsheetUp) {
```
Comment (TammiLion) [inline lib/features/family/features/profiles/screens/profile_selection_screen.dart:97]: I know you asked if there was a better solution for this during the standup. I'll take a look at that outside of this PR.

**Review by TammiLion — APPROVED** [review]
Just some small comments.

File: lib/features/family/features/profiles/cubit/profiles_state.dart (hunk 6 above)
Comment (Daniela510) [reply lib/features/family/features/profiles/cubit/profiles_state.dart:7]: I do use the data in ProfilesNotSetupState, and this comment revealed  a bug to me that i just fixed, thanks!

File: lib/features/family/features/profiles/cubit/profiles_cubit.dart (hunk 5 above)
Comment (Daniela510) [reply lib/features/family/features/profiles/cubit/profiles_cubit.dart:52]: do you mean something like:
```
(members) {
        _emitCachedMembers(members);
      },
```
  ?

**Pull request merged**